### ISSUE TITLE
Caption and browser out of sight at the bottom

### ISSUE THREAD

**Issue by p-p-s** [umap-project/umap#2140]
**Describe the bug**
When a map is embedded into a Web page, depending on the space available for the map, the caption as well as the browser pane is **unnecessarily displayed at the bottom of the map covering the entire space from the left edge to the right edge**. This can be totally **out of sight** without scrolling down, and when scrolling down it is easily perceived as something **not belonging to the map**.

You may not view this as a bug, rather as an enhancement request. But definitely it is a serious usability issue, especially for newbie uMap viewers. Even for me, who edited uMap maps before and was familiar with the functionality of the buttons, when I embedded the map in a Web site that grants about 2/3 of the overall width to the map, when checking it in a full-size browser window on a 1920 x 1080 screen, I thought the caption would not be shown and clicking the layers icon to bring up the browser would also not work. **It took me time to realize that the caption pane was not missing**, but at the bottom of the map, fully out of sight. When I scrolled down and the pane started to become visible, I first thought the text on white background that started appearing was something following the map on the Web page, not part of the map. **Many Web site visitors may already have walked away** because of the impression "it's not working".

This is especially annoying as uMap gives the **pane far more space than needed**. The following screenshot demonstrates that the caption pane would have easily fit onto the map close to the upper left corner. In a normal window it is already very generous with white space at the layer names, maybe as it defines the size of the pane to support maximum possible layer name lengths rather than sizing for the currently given layer names.
![image](https://github.com/user-attachments/assets/1ad03675-ae6c-481f-8865-10823c6eb1d0)

**To Reproduce**
See the appearance at https://sesfkhandbari.org/new/tour/shiva-dhara-trek/#map
uMap instance: https://umap.openstreetmap.fr as of 16 Sep 2024
Map used: https://umap.openstreetmap.fr/en/map/makalu-base-camp-trek_1112563 (with this direct link it displays as it shall do)
Integrated in the WordPress scheme Travel Tour by Good Layers using the Sidebar Wrapper design, adding the map with the following tag:
`<code><iframe src="//umap.openstreetmap.fr/en/map/makalu-base-camp-trek_1112563?scaleControl=true&amp;miniMap=false&amp;scrollWheelZoom=true&amp;zoomControl=true&amp;editMode=disabled&amp;moreControl=false&amp;searchControl=null&amp;tilelayersControl=null&amp;embedControl=null&amp;datalayersControl=true&amp;onLoadPanel=caption&amp;captionBar=false&amp;captionMenus=true" width="100%" height="1200px" frameborder="0" allowfullscreen="allowfullscreen"></iframe></code>`

**Expected behavior**
1. The caption pane (as well as the browser pane) should **in any case be shown close to the upper left corner** of the map, right to the menu. Let it cover as much space as it needs, but just keep a tiny stripe of map visible to the right of the caption (browser) pane. That will make instantly clear that it is a pane on top of the map and that it can be closed to make the map fully visible. 
2. Ideally, **size the caption/browser pane width for the actual current layer names**. 
3. Should you for any reason don't want to place caption and browser anyway close to the upper left corner, give the uMap editors a **checkbox that allows them to force placement of the panes close to the upper left corner**, accepting that part of the layer names may be clipped off on tiny screen spaces.  
 
**Desktop:**
 - OS: Windows 10 
 - Screen resolution: 1920 x 1080 
 - Edge, 128.0.2739.67 (Official build) (64-bit), fill-size window

**Smartphone:**
 - Device: Xiaomi Mi 8 SE
 - OS: Android 9
 - MIUI 10.3 by xiaomi.eu 8.12.27 Beta 
 - Browser: Chrome, 128.0.6613.128

**Additional context**
 - WordPress scheme Travel Tour by Good Layers, updated to the current version
 - Page design based on the "Sidebar Wrapper" template, placing the map in the wider column

**Comment by davidbgk** [comment]
The map located at https://sesfkhandbari.org/new/tour/shiva-dhara-trek/#map doesn't look to have the correct link (https://umap.openstreetmap.fr/en/map/makalu-base-camp-trek_1103423 vs. https://umap.openstreetmap.fr/en/map/makalu-base-camp-trek_1112563 which leads to a 404).

I don't understand why you put the iframe within a `<code></code>` HTML tag?

**Comment by p-p-s** [comment]
Hi David, **sorry for the link which did not work for you**. I should have tested the effective publishing in a different browser. If you try it now, these links should now work:
- https://sesfkhandbari.org/new/tour/shiva-dhara-trek/#map
- https://sesfkhandbari.org/new/tour/makalu-base-camp-trek/#map

Regarding your question about the <code></code> HTML tag: I don't recall anymore whether it was the Good Layers' Travel Tour theme itself that entered already <code></code> in the Text area for the element or I thought, I should click the "code" button there because I want to enter the iframe tag which is HTML code, or whether any Good Layers documentation told to do this. **Anyway, I now removed the <code></code> around the iframe tag, which works, but does not make any difference**. The second link above is now without <code></code>:
![image](https://github.com/user-attachments/assets/760a18d9-6737-421b-a527-bdec95c938c3)

Please, check the situation on a PC and also on a mobile phone. The following screenshot is from a Xiaomi MI 8 SE, Android version 9, MIUI 10.3 in an updated Chrome browser:
![image](https://github.com/user-attachments/assets/0fc0ddfb-a3b9-4394-9cc1-b26076231fca)

I thought already that you might remove the bug label. But this is a **serious usability issue**. Compared to this one, the three other tickets which I opened are all peanuts. 

To illustrate the usability issue: I am supporting a non-profit Makalu Arun Social Trek organization in Nepal. The links above point into their test website. As you can see they are able to maintain a large website based on a WordPress theme. But the guy I am coaching **was not able to reach the opened caption pane to activate another layer**. He did not understand that he would need to take the cursor out of the map in order to scroll down to the caption pane. I was with him in a Google Meet session and tried to make him understand that he needs to move the cursor out of the map in order to be able to scroll down. He did not understand. Instead he was panning and panning and zooming in and out. This was the case **for 15 minutes while I was trying to instruct him verbally**, seriously. But this person is able to maintain a rather complex WordPress site, i.e. he **is somewhat IT-savvy**. For me this means that **average users will also not get to the Caption pane at the bottom out of sight. They will never activate any other layer and will judge that the respective icons in the left menu just do not work - and leave the website**.

**Comment by yohanboniface** [comment]
Can you summarize the issue in one sentence, without bold, please ?

**Comment by p-p-s** [comment]
On mobile phones and on PCs in spaces smaller than a full webpage, caption as well as browser pane appear at the bottom, out of sight and many people will not understand how to navigate to it and that there are more layers to activate. Even people who know how to scroll down will often think the caption or browser does not anymore belong to the map as it appears as text following the map.

**Comment by yohanboniface** [comment]
Thanks, this is clearer and more actionnable. Though I'm not totally convinced it's a pure uMap issue.  For example, in your site, the page background and the map panels are of same color. Maybe distinguish them would help (or adding a border to the iframe) ? Maybe  decreasing the iframe height also (so the left controls remain visible) ?

@Aurelie-Jallut thoughts ?

**Comment by Aurelie-Jallut** [comment]
If the problem is that the panel seems to be separate from the map, could we reduce its width a little to allow the map to be seen on the left and right?

(Visually losing the left-hand buttons seems more problematic to me).

**Comment by p-p-s** [comment]
Hi uMap dev team,
Has the project taken any decision? 
For instance, implementing an option that allows map developers to force caption and browser to the standard position adjacent to the left-hand buttons, close to the upper left corner - on their own risk of clipping parts of their layer titles?
Thanks
Peter

**Comment by yohanboniface** [comment]
Have you tried adding a border to the iframe ?

**Comment by p-p-s** [comment]
Hi Yohan,

No, I did not. Because in my eyes this would be a work-around effectively stimulating only very few of our customers, my guessing 10% of them, while for 90% of them the caption (or browser) pane would still just be out of sight, and for them pressing menu buttons on the left side would result in nothing, map not working, website not working, provider not reliable, let's look for other trekking providers. 

But I want my trekking provider to succeed, because it is the only non-profit trekking provider in Nepal among hundreds of commercial trekking providers. The social impact of my client on local communities is very important as Nepal is such a poor country with almost no social welfare. They make possible that orphans in East Nepal receive basic school education who would otherwise end up as illiterates. This is why I put much voluntary effort in their support and I would love if the uMap project team would support me in that. I strongly believe that proper uMap maps with layers as designed would catapult them to the top of the competition. Please help.

-Peter

**Comment by p-p-s** [comment]
I systematically tested out what happens when varying the height for the map at the iframe setup.

I tested heights of 1200px, 1000px, 800px, 600px, 450px, 400px, through 350px on three different devices: a typical mobile phone, an iPad Mini, and a laptop, as also the used WordPress scheme, Good Layers Travel Tour, a very popular choice of tracking providers, as many other WordPress sites controls the appearance according to the three categories mobile phone, tablet, laptop/desktop. I took a screenshot at every device at all heights and arranged them in two-dimensional matrix for best overview. Please have a look at [uMap Caption Placement by Height_96.xlsx](https://www.dropbox.com/scl/fi/eejtoml2ec2hesct4ti5k/uMap-Caption-Placement-by-Height_96.xlsx?rlkey=ykfmwi61a1eofmox2erbqdkvh&st=fsth60c8&dl=0).

It seems to me that uMap is placing the caption/browser in the standard position close to the left-hand buttons if the map space is in landscape format and otherwise at the bottom of the map using the entire available width. This was more reasonable at times when only PCs were using the Web, although in many cases an awfully wide caption/browser box does not make any sense, and sometimes the map content suggests portrait format rather than landscape format. 

But today the most common clients are mobile phones, not any more PCs. In addition there are few tablets. And phones and tablets are almost always held in portrait mode. You may now want to advise that mobile phone and tablet users should view the map with the device held in landscape mode. But most users never put their devices in landscape mode. Look at Instagram and TikTok. Mountain panoramas in portrait mode, seascapes with huge boring skies! A precondition for tilting the screen would be that people would at least guess that this would improve their experience with the map. But who would expect something to come into sight what is out of sight and they have no clue that it does exist? You may also argue, users should go full-screen. Indeed it improves the experience with the map. But I assume that for most customers of the trekking provider I serve, uMap is completely new. Some will just try out a button with an icon they are not familiar with, others will not. 

Please consider best supporting portrait mode devices, either by changing the automatic positioning decision by taking an aspect ratio around 2:1 as the threshold, or give map designers the option to enforce the placement of caption and browser aside the upper left icons. I am convinced this will help also many other uMap user projects. 

Thanks for creating this wonderful mapping service! Nothing compares to uMap!

**Comment by yohanboniface** [comment]
There is certainly room for improvement for panels in portrait mode, and/or for better controls of the default position, but I don't see a clear and obvious and not too costly fix atm (maybe @Aurelie-Jallut does), so it will take some time to get there.
In the meantime, maybe try to set a smaller height (eg. 600 px instead of 1200), and add a "See full screen" button ?

![Image](https://github.com/user-attachments/assets/33c6ef05-dd80-4726-9c3c-8f43d1d191ac)

**Comment by Aurelie-Jallut** [comment]
Adding a margin so that the map can be seen would make it possible to visually associate the legend with the map. 

Set a condition for display at a given resolution (media queries): Reduce the height of the iframe

![Image](https://github.com/user-attachments/assets/f9515551-fcea-4d52-b936-9fcd8ea5a814)

**Comment by p-p-s** [comment]
> Set a condition for display at a given resolution (media queries): Reduce the height of the iframe

Hi Aurelie-Jallut,
I'm not sure I understand your suggestion. Maybe I'm too stupid. 
I am not aware of options to set the height conditionally, based on the available devices' screen height or anything, when it comes to WordPress schemes.
**I was going now for a compromise height of 600px across all devices** which would make part of the caption visible an encourage scrolling down to see it completely. It would look like the screenshots in the 600px row in [uMap Caption Placement by Height_96.xlsx](https://www.dropbox.com/scl/fi/eejtoml2ec2hesct4ti5k/uMap-Caption-Placement-by-Height_96.xlsx?rlkey=ykfmwi61a1eofmox2erbqdkvh&st=fsth60c8&dl=0) - with captions, where placed at the bottom, running from the left edge to the right edge of the map's space.
**But your screenshot is much better: there is a small gap between the caption panel and the map's edge to the left and to the right. I like that. How do you manage to get that?**
Thanks
Peter

**Comment by Aurelie-Jallut** [comment]
It's un mock-up to test the proposal : )

**Comment by p-p-s** [comment]
Hi uMap team,
Do you have a decision for this one?
Thanks
Peter
